### PR TITLE
Fix dev container

### PR DESCRIPTION
**Pull request by amirmolavi**
**Summary of Changes for Dev Container Setup**

I made several updates to the dev container to resolve issues I encountered during setup. Based on my experience, these changes should make the dev container functional for everyone. My understanding is that the current setup may not be fully functional; however, I could be mistaken due to my limited expertise in this area. Below is a summary of the changes:

1) Added `git` and `git-lfs` Installation: Included `git` to support `pre-commit` installation and `git-lfs` to manage large files in the repository stored via `Git LFS`.

Updated `pixi.lock`: The existing `pixi.lock` file was outdated, so I refreshed it to match the current dependencies.

Added `linux-aarch64` Architecture: This architecture was added to the `pyproject.toml` to ensure compatibility with all target platforms.

Set Executable Paths to `/.pixi/bin`: Updated `devcontainer.json` and `Dockerfile` to use `/.pixi/bin` for executables since the project is installed at `/. ` Paths for the interpreter and commands have been adjusted accordingly.

Fixed `.pre-commit-config.yaml`: Updated the pre-commit configuration to install the project in `/.` instead of `/workspaces/llamabot`, as this was causing commit failures.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -7,9 +7,9 @@
     "customizations": {
         "vscode": {
             "settings": {
-                "python.defaultInterpreterPath": "/opt/conda/envs/llamabot",
+                "python.defaultInterpreterPath": "/.pixi/envs/default/bin/python",
```
Comment (ericmjl) [inline .devcontainer/devcontainer.json:10]: great catch!

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -31,6 +31,6 @@
     "forwardPorts": [
         8888
     ],
-    "postCreateCommand": "/opt/conda/envs/llamabot/bin/pre-commit install && /opt/conda/envs/llamabot/bin/python -m ipykernel install --user --name llamabot",
+    "postCreateCommand": "/.pixi/envs/default/bin/pre-commit install && /.pixi/envs/default/bin/python -m ipykernel install --user --name llamabot",
```
Comment (ericmjl) [inline .devcontainer/devcontainer.json:34]: likewise here!

**Review by ericmjl — APPROVED** [review]
I approve! I think we just have to resolve merge conflicts and we're good to go!

**Comment by ericmjl** [comment]
@amirmolavi I'm going to give you direct collaborator access to the repo, I think this'll help speed up the cycle time. You'll be able to avoid waiting on me to approve runs on CI/CD (I think).

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -47,7 +47,7 @@ repos:
     hooks:
       - id: pixi-install
         name: pixi-install
-        entry: pixi install
+        entry: pixi install --manifest-path /pyproject.toml
```
Comment (ericmjl) [inline .pre-commit-config.yaml:50]: Ah, just saw the CI logs -- I think this one will be an issue b/c of the need for pre-commit hooks to work both within a devcontainer and through a local pixi installation. What are your thoughts on the best way to support both? Move back to $CWD/pyproject.toml?

Comment (amirmolavi) [reply .pre-commit-config.yaml:50]: That would not work. If I just change this back then pre-commit will fail locally and I cannot commit. Let ma take a closer look into thsi.

**Comment by ericmjl** [comment]
I think the final conflicts that we're observing here might be due to the latest changes that I merged in. It's something I observed empirically using pixi. Could you see if applying changes from the latest `main` branch might help resolve those conflicts please, @amirmolavi?

File: .pre-commit-config.yaml (hunk 3 above)
Comment (amirmolavi) [reply .pre-commit-config.yaml]: @ericmjl I messed up several things in my local permissions, I am gonna close this PR. reclone and then reopen it, sorry for the inconvenience,

**Pull request closed without merging**